packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -212,7 +207,10 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
     }
 
     public async isRoot(): Promise<boolean> {
-        return (await this.getInitialSnapshotDetails()).isRootDataStore;
+        // This call updates this.isRootDataStore if it has not yet been updated
+        // The initial value is stored in the initial snapshot of the data store
+        await this.getInitialSnapshotDetails();
+        return this.isRootDataStore;
```

Comment:
I find it dangerous to have this implicit dependency that is not enforced in any way that getInitialSnapshotDetails() should be called first.
At minimum, I'd make this.isRootDataStore to be boolean | undefined, and thus ensure that compiler helps us catch places where someone accesses this.isRootDataStore outside this function

Reply:
Other way to deal with it - make sure that getInitialSnapshotDetails() returns pkg & isRootDataStore, and remove this.isRootDataStore - based on usage it sounds like it should be easy fix.

Reply:
Let's fix this properly in main as we need this fix to go out asap. I created an issue to track this - https://github.com/microsoft/FluidFramework/issues/9355.

Reply:
getInitialSnapshotDetails() was returning pkg and isRootDataStore but that seems wrong now because isRootDataStore is not a property of the initial snapshot anymore. We can rename it to something like data store attributes and then it might make sense for isRootDataStore to be updated later on.